src/MessagingThreadHistory/MessagingThreadHistory.tsx

```diff
@@ -48,11 +48,21 @@ export const MessagingThreadHistory = React.forwardRef(
     { className, threadID, messages }: Props,
     containerRef: React.MutableRefObject<HTMLDivElement>,
   ) => {
+    // ----------- Scroll position references
+
+    // Reference attached to the last message's DOM element
     const lastMessageRef = useRef<HTMLDivElement>(null);
+
+    // Maintains a reference as to what the last (highest) messages index was on the last render
     const lastMessageIndex = useRef(
       messages.length > 0 ? messages[messages.length - 1].index : null,
     );
-    const messagesWithDividers = _interleaveMessagesWithDividers(messages, lastMessageRef);
+
+    // Determines whether the user was scrolled to the bottom before we potentially
+    //  insert new messages at _interleaveMessagesWithDividers()
+    const isScrolledToBottom = calcIsScrolledToBottom(containerRef);
```

Comment:
The main difference is I calculate this value before `useLayoutEffect`, as that gets fired after the whole render function is run and the new messages are appended to the bottom of the thread history.